### PR TITLE
Bug fix: Assure that feature dataframe indices are unique

### PR DESCRIPTION
**Pull request by JuliaKukulies**
The feature detection does currently not work when  `min_distance` is not 0. An exception is raised in `filter_min_distance()` , because the input feature dataframe for this function can contain index values that are double-defined as a consequence of  concatenating dataframes in `feature_detection_multithreshold_timestep()`. This PR solves #184 by assuring that the feature dataframe that is created in `feature_detection_multithreshold_timestep()` gets unique indices. 

I also added a test for  `filter_min_distance()` to test if features are removed as expected within a specified minimum distance and I removed redundant imports from `test_feature_detection.py`. 

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

File: tobac/tests/test_feature_detection.py [hunk 1]
```diff
@@ -53,6 +52,71 @@ def test_feature_detection_multithreshold_timestep(
     assert fd_output.iloc[0]["hdim_2"] == pytest.approx(test_hdim_2_pt)
 
 
+@pytest.mark.parametrize(
+    "test_threshs, min_distance, dxy", [([1, 2, 3], 100000, 10000)]
+)
+def test_filter_min_distance(test_threshs, min_distance, dxy):
+    """
+    Tests ```tobac.feature_detection.filter_min_distance
```
Comment (freemansw1) [inline tobac/tests/test_feature_detection.py:60]: needs closing ```

**Review by freemansw1 — APPROVED** [review]
Great work and glad that we can quickly address #184!

**Review by snilsn — APPROVED** [review]
Good job @JuliaKukulies ! Nothing to complain about in this quick bug fix and great that the code coverage is increased at the same time!

**Comment by JuliaKukulies** [comment]
Thanks for the quick review! I will merge after #186 has been merged
